### ISSUE TITLE
[FEA] Regexp: Support octal digits in regexp pattern

### ISSUE THREAD

**Issue by andygrove** [NVIDIA/spark-rapids#4288]
**Is your feature request related to a problem? Please describe.**
We currently fall back to CPU for regexp patterns containing octal digits due to cuDF producing incorrect results.

The issue is demonstrated by this test in `RegularExpressionTranspilerSuite`:

```scala
ignore("known issue - octal digit") {
  val pattern = "a\\141|.$" // using hex works fine e.g. "a\\x61|.$"
  assertCpuGpuMatchesRegexpFind(Seq(pattern), Seq("] b["))
}
```

**Describe the solution you'd like**
Support on GPU consistently with Spark.

**Describe alternatives you've considered**
None

**Additional context**
None

**Comment by andygrove** [comment]
In Java, octal digits must start with `\\0` and the parser currently does not implement this logic, so this is likely the cause of the test failure.

**Comment by jlowe** [comment]
Fixed by #5645.
